pkg/proxy/proxy.go

```diff
@@ -186,10 +187,14 @@ type invocationError struct {
 
 func writeError(writer http.ResponseWriter, err error, accept string) {
 	accepts := goautoneg.ParseAccept(accept)
-	preferJSON := accepts[0].Type == "application" && accepts[0].SubType == "json"
-
-	if preferJSON {
-		writer.Header().Set("content-type", "application/json")
+	preferJSON := false
+	if len(accepts) != 0 {
```

Comment:
nit: this could be simplified to:
```
if len(accepts) > 0 && accepts[0].Type == "application" && accepts[0].SubType == "json" {
   writer.Header().Set("content-type", "application/json")
} else {
   writer.Header().Set("content-type", "text/plain")
}
```

Reply:
that's a fair suggestion, but this line is already on the main branch. I'm not sure why GitHub is showing it as new.

@cz9779 can you rebase this PR on top of the main branch so that the diff is cleaner?

Reply:
`preferJSON` is actually needed a few lines later, I withdraw my initial comment

Reply:
@fbiville Thanks for this suggestion but the var preferJSON is used again a several lines down at L#211, so it's set by the conditions here at L#192...